### PR TITLE
Sprint 45

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 45

- #5665 Dev Team: Document manual tool scripts
- #5381 Docket Clerk: Provide “other” filing party during QC and paper filing
- #5815 Update Correspondence File design
- #4295 Docket Clerk: Service for paper-scanned documents
- #5811 Serve a court-issued and paper-filed document that was added to the Docket Record
- #5816 Data Migration: Legacy Practitioners representing “Other Petitioners,” participants, and/or intervenors
- #5814 Require a signature for Orders and Decisions being added to the docket record + no manual signatures for notices
- #5812 Applying and editing signatures on uploaded PDF drafts
- #5810 Update message boxes
- #5666 Dev Team: Allow production domain configuration (Groom w/18F) In Progress
- #4533 Ability to easily reference case details while generating an order and QCing In Progress

### CODE REVIEW

File: docs/entities/Case.md [hunk 1]
```diff
@@ -439,18 +427,16 @@
...
+        description: "If this case is consolidated, this is the docket number of the lead case. It is the lowest docket number in the consolidated group."
       rules: 
         - 
-          name: "guid"
+          name: "pattern"
           args: 
-            options: 
-              version: 
-                - "uuidv4"
+            regex: "/^([1-9]\\d{2,4}-\\d{2})$/"
```
Comment (mmarcotte) [inline docs/entities/Case.md:439]: sweet; makes me wonder whether this will have to change when we have all of the cases in the system.

Comment (ericsorenson) [reply docs/entities/Case.md:439]: Using a UUID4? From wikipedia: `the probability to find a duplicate within 103 trillion version-4 UUIDs is one in a billion`

Comment (mmarcotte) [reply docs/entities/Case.md:439]: err, i meant the docket 1234-12
```
 regex: "/^([1-9]\\d{2,4}-\\d{2})$/"
```

Comment (mmarcotte) [reply docs/entities/Case.md:439]: I think I get it, though, 99999 max cases in a 2-digit year.

File: web-client/src/presenter/computeds/messageDocumentHelper.test.js [hunk 2]
```diff
@@ -290,6 +290,35 @@ describe('messageDocumentHelper', () => {
...
+        caseDetail: {
+          correspondence: [],
+          docketRecord: [],
+          documents: [
+            {
+              documentId: '123',
+              entityName: 'Document',
+              eventCode: 'SDEC',
+              isDraft: true,
+              signedAt: '2020-06-25T20:49:28.192Z',
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/messageDocumentHelper.test.js:308]: would a draft document have a signedAt?

File: web-client/src/presenter/computeds/messageDocumentHelper.js [hunk 3]
```diff
@@ -52,6 +54,9 @@ export const messageDocumentHelper = (get, applicationContext) => {
     caseDocument &&
     caseDocument.eventCode === INITIAL_DOCUMENT_TYPES.petition.eventCode;
 
+  const isStipulatedDecision =
+    caseDocument && caseDocument.eventCode === STIPULATED_DECISION_EVENT_CODE;
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/messageDocumentHelper.js]: Seems like if we have an empty `caseDocument` object, we should be doing something about that sooner, which would allow the rest of this function to be more concise.

File: web-client/src/presenter/computeds/messageDocumentHelper.js [hunk 4]
```diff
@@ -99,6 +108,11 @@ export const messageDocumentHelper = (get, applicationContext) => {
   const showServePetitionButton =
     showNotServed && isPetitionDocument && permissions.SERVE_PETITION;
 
+  const showSignStipulatedDecisionButton =
+    isInternalUser &&
+    caseDocument.eventCode === PROPOSED_STIPULATED_DECISION_EVENT_CODE &&
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/messageDocumentHelper.js:112]: And I think we already have this with `isStipulatedDecision`

File: shared/src/business/useCases/pendingItems/generatePrintablePendingReportInteractor.js [hunk 5]
```diff
@@ -24,13 +24,14 @@ exports.generatePrintablePendingReportInteractor = async ({
     throw new UnauthorizedError('Unauthorized');
   }
 
+  //TODO: prefereably decodeURIComponent in the router/utility method for easy mocking
```
Comment (julialeague) [inline shared/src/business/useCases/pendingItems/generatePrintablePendingReportInteractor.js:27]: is there a related card in the backlog for this work?

File: web-client/src/presenter/computeds/caseDetailHeaderHelper.js [hunk 6]
```diff
@@ -48,11 +48,17 @@ export const caseDetailHeaderHelper = (get, applicationContext) => {
     }
   }
 
-  const showConsolidatedCaseIcon = !!caseDetail.leadCaseId;
+  const showConsolidatedCaseIcon = !!caseDetail.leadDocketNumber;
 
   const showCreateOrderButton = permissions.COURT_ISSUED_DOCUMENT;
 
-  const showAddDocketEntryButton = permissions.DOCKET_ENTRY;
+  const hasDocketEntryPermission = permissions.DOCKET_ENTRY;
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/caseDetailHeaderHelper.js]: perhaps added for debugging, but it doesn't live very long

File: web-api/migrations/00018-eligible-for-trial-docket-number.test.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
+const { forAllRecords } = require('./utilities');
+const { up } = require('./00018-eligible-for-trial-catalog-docket-number');
```
Comment (julialeague) [inline web-api/migrations/00018-eligible-for-trial-docket-number.test.js]: This test file name is missing the word "catalog" in it, per the testee filename.

File: web-client/integration-tests/adcSignsProposedStipDecision.test.js [hunk 8]
```diff
@@ -45,8 +44,8 @@ describe('Sr. Attorney Signs Proposed Stipulated Decision', () => {
     test,
     'Jeff, this is ready for review and signature',
   );
-  docketClerkVerifiesStipulatedDecisionExistsInInbox(
-    test,
-    'Donna, this is not ready to serve. I need to follow up on something first',
-  );
+  // docketClerkVerifiesStipulatedDecisionExistsInInbox(
```
Comment (julialeague) [inline web-client/integration-tests/adcSignsProposedStipDecision.test.js]: Should there be a note about why this is commented out?

**Comment by julialeague** [comment]
- [x] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [x] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/324/head:sprint-45`), and then switch to that branch (e.g. `git checkout sprint-45`)
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [x] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
	- [x] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/src/presenter/computeds/messageDocumentHelper.test.js (hunk 2 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/messageDocumentHelper.test.js:308]: It can, but draft documents can't be served.

File: web-client/src/presenter/computeds/messageDocumentHelper.js (hunk 3 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/messageDocumentHelper.js]: Good point, I think that was a bit of unnecessary defensive coding.

File: web-client/src/presenter/computeds/messageDocumentHelper.js (hunk 4 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/messageDocumentHelper.js:112]: This is `PROPOSED_STIPULATED_DECISION_EVENT_CODE` vs `STIPULATED_DECISION_EVENT_CODE`.

File: shared/src/business/useCases/pendingItems/generatePrintablePendingReportInteractor.js (hunk 5 above)
Comment (ericsorenson) [reply shared/src/business/useCases/pendingItems/generatePrintablePendingReportInteractor.js:27]: Yes! https://trello.com/c/BcRg2BzP

File: web-api/migrations/00018-eligible-for-trial-docket-number.test.js (hunk 7 above)
Comment (ericsorenson) [reply web-api/migrations/00018-eligible-for-trial-docket-number.test.js]: Good catch!

File: web-client/integration-tests/adcSignsProposedStipDecision.test.js (hunk 8 above)
Comment (ericsorenson) [reply web-client/integration-tests/adcSignsProposedStipDecision.test.js]: I'll delete it, it's no longer needed.

File: web-client/src/presenter/computeds/caseDetailHeaderHelper.js (hunk 6 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/caseDetailHeaderHelper.js]: Good catch!

File: web-client/integration-tests/adcSignsProposedStipDecision.test.js (hunk 8 above)
Comment (ericsorenson) [reply web-client/integration-tests/adcSignsProposedStipDecision.test.js]: Updated!

File: web-api/migrations/00018-eligible-for-trial-docket-number.test.js (hunk 7 above)
Comment (ericsorenson) [reply web-api/migrations/00018-eligible-for-trial-docket-number.test.js]: Updated!

File: web-client/src/presenter/computeds/caseDetailHeaderHelper.js (hunk 6 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/caseDetailHeaderHelper.js]: Updated!

File: web-client/src/presenter/computeds/messageDocumentHelper.js (hunk 3 above)
Comment (ericsorenson) [reply web-client/src/presenter/computeds/messageDocumentHelper.js]: Updated!

**Review by julialeague — APPROVED** [review]
@mmarcotte, I recommend you accept this pull request into the staging environment. Thanks!

File: web-client/src/presenter/computeds/messageDocumentHelper.js (hunk 3 above)
Comment (mmarcotte) [reply web-client/src/presenter/computeds/messageDocumentHelper.js]: thank you!
